tedana/decomposition/gift_pca.py

```diff
@@ -0,0 +1,652 @@
+        xc = xc * parzen_window_3D
+        xf = fftshift(fftn(xc))
+
+    else:
+        raise ValueError('Unrecognized matrix dimension.')
+
+    xf = abs(xf)
+    xf[xf < 1e-4] = 1e-4
+
+    # Estimation of the entropy rate
+    out = 0.5 * np.log(2 * np.pi * np.exp(1)) + _sumN(np.log(abs(
+        (xf)))) / 2 / _sumN(abs(xf))
+
+    return out
+
+
+def _est_indp_sp(x):
+    """
+    Estimate the effective number of independent samples based on the maximum
+    entropy rate principle of stationary random process.
+
+    Parameters
+    ----------
+    x : ndarray
+        The data to have the number of samples estimated
+
+    Returns
+    -------
+    s : int
+        Number of iterations required to estimate entropy rate
+    entrate_m : float
+        The entropy rate of the data
+
+    Notes
+    -----
+    TOOD: explain math
+
+    References
+    ----------
+    TODO: add references
+    """
+
+    dimv = x.shape
+    s0 = None
+
+    for j in range(np.min(dimv) - 1):
+        x_sb = _subsampling(x, j + 1)
+        entrate_m = _entrate_sp(x_sb, 1)
+
+        ent_ref = 1.41
+        if entrate_m > ent_ref:
+            s0 = j
+            break
+
+    if not s0:
+        raise ValueError('Ill conditioned data, can not estimate'
+                         'independent samples.')
+    s = s0
+    LGR.debug('Estimated the entropy rate of the Gaussian component '
+              'with subsampling depth {}'.format(j))
+
+    return s, entrate_m
+
+
+def _subsampling(x, s):
+    """
+    Subsampling the data evenly with space 's'.
+
+    Parameters
+    ----------
+    x : ndarray
+        The data to be subsampled
+    s : int
+        The subsampling depth
+
+    Returns
+    -------
+    out : ndarray
+        Subsampled data
+    """
+
+    # First index from which to start subsampling for each dimension
+    x0 = [0, 0, 0]
+    n = x.shape
+
+    if x.ndim == 3 and np.min(n) != 1:  # 3D
+        out = x[np.arange(
+            x0[0], n[0], s), :, :][:, np.arange(
+                x0[1], n[1], s), :][:, :, np.arange(x0[2], n[2], s)]
+    else:
+        raise ValueError('Unrecognized matrix dimension!(subsampling)')
+
+    return out
+
+
+def _kurtn(x):
+    """
+    Normalized kurtosis funtion so that for a Gaussian r.v. the kurtn(g) = 0.
+
+    Parameters
+    ----------
+    x : ndarray
+        The data to calculate the kurtosis of
+
+    Returns
+    -------
+    kurt : (1:N) array-like
+        The kurtosis of each vector in x along the second dimension. For
+        tedana, this will be the kurtosis of each PCA component.
+    """
+
+    kurt = np.zeros((x.shape[1], 1))
+
+    for i in range(x.shape[1]):
+        a = detrend(x[:, i], type='constant')
+        a = a / np.std(a)
+        kurt[i] = np.mean(a**4) - 3
+
+    kurt[kurt < 0] = 0
+
+    return kurt
+
+
+def _icatb_svd(data, numpc=None):
+    """
+    Run Singular Value Decomposition (SVD) on input data and extracts the
+    given number of components (numpc).
```

Comment:
I think the rest of the codebase has this as `n_comps`. Can we standardize ?

Reply:
Done